### PR TITLE
[WIP] OWFile: Enable open file from url

### PR DESCRIPTION
**Pull request by VesnaT**
Accordingly, data file combo box is now editable.

### CODE REVIEW

File: Orange/widgets/data/owfile.py [hunk 1]
```diff
@@ -21,6 +21,14 @@ def add_origin(examples, filename):
             var.attributes["origin"] = dir_name
 
 
+def is_url_valid(url):
+    try:
+        urllib.request.urlopen(url)
+        return True
+    except:
```
Comment (kernc) [inline Orange/widgets/data/owfile.py]: The [documentation](https://docs.python.org/3.4/library/urllib.request.html#module-urllib.request) says it raises `URLError` on errors?

Comment (VesnaT) [inline Orange/widgets/data/owfile.py]: Should handle urllib.error.HTTPError as well.

Comment (kernc) [inline Orange/widgets/data/owfile.py]: https://docs.python.org/3.4/library/urllib.error.html#urllib.error.HTTPError

File: Orange/widgets/data/owfile.py [hunk 2]
```diff
@@ -291,8 +325,19 @@ def browse_file(self, in_demos=0):
             self.recent_paths.remove(recent)
 
         self.recent_paths.insert(0, recent)
-        self.set_file_list()
-        self.open_file(self.recent_paths[0].abspath)
+
+    @staticmethod
+    def is_url_valid(url):
+        try:
+            return bool(Table.from_url(url)), ""
```
Comment (janezd) [inline Orange/widgets/data/owfile.py]: This test is too expensive if data is large. I think we should just check whether the link works and close the connection without transferring any data.

Comment (kernc) [inline Orange/widgets/data/owfile.py]: why not just `return True, Table(...)`?

**Comment by janezd** [comment]
Additional changes needed:

- nicer error reporting when the URL is invalid or contains invalid data
- support for Google sheets
- standard icons for files and URLs

**Comment by VesnaT** [comment]
Support for Google sheets is planned for another pull request.

**Comment by VesnaT** [comment]
Support for Google sheets is planned for another pull request.

**Pull request merged**